charm-slurmctld/src/etcd_ops.py

```diff
@@ -87,15 +87,75 @@ def is_active(self) -> bool:
             logger.error(f'## Could not check etcd: {e}')
             return False
 
-    def configure(self):
-        """Configure etcd instance."""
-        # TODO set up password?
-        # TODO set up port?
-        # TODO enable metrics? Relate to prometheus?
-        logger.debug("## configuring etcd")
+    def configure(self, root_pass):
+        """Configure etcd service.
 
-    def set_list_of_accounted_nodes(self, nodes: List[str]) -> None:
+        :root_pass: password to use for root account
+        """
+        logger.debug("## configuring etcd")
+        self.start()
+
+        # some configs can only be applied with the server running
+        self.setup_default_roles(root_pass)
+
+    def setup_default_roles(self, root_pass) -> None:
+        """Set up default etcd roles.
+
+        We use three roles:
+        - root: for slurmctld operations
+            - has full r/w permissions
+        - slurmd: for slurmd charms
+            - has r/w permissions only for nodes/* keys
+        - munge: for external accounts reading the munge key
+            - has r permissions for munge/* keys
+        """
+        logger.debug("## creating default etcd roles/users")
+        cmds = [# create root account with random pass
+                f"etcdctl user add root:{root_pass}",
+                # create root role
+                "etcdctl role add root",
+                "etcdctl user grant-role root root",
+
+                # create slurmd user
+                "etcdctl user add slurmd:slurmd",
+                # create slurmd role
+                "etcdctl role add slurmd",
+                "etcdctl role grant-permission slurmd readwrite --prefix=true nodes/",
+                # grant slurmd user the slurmd role
+                "etcdctl user grant-role slurmd slurmd",
+
+                # create munge role
+                "etcdctl role add munge-readers",
+                "etcdctl role grant-permission munge-readers read --prefix=true munge/",
+
+                # enable auth
+                "etcdctl auth enable",
+               ]
+
+        for cmd in cmds:
+            logger.debug(f"## etcdctld: {cmd}")
```

Comment:
This log line feels like it could be clearer, like:
```suggestion
            logger.debug(f"## executing command: {cmd}")
```